modules/cnf-reducing-netqueues-using-pao.adoc

```diff
@@ -0,0 +1,346 @@
+* `vendorID`: Network device vendor ID represented as a 16 bit hexmadecimal number with a 0x prefix.
+* `deviceID`: Network device ID (model) represented as a 16 bit hexmadecimal number with a 0x prefix.
++
+[NOTE]
+====
+When a `deviceID` is specified, the `vendorID` must also be defined. A device that matches all of the device identifiers specified in a device entry `interfaceName`, `vendorID` or a pair of `vendorID` plus `deviceID` representing a network device qualifies as a network device that will have its net queues count set to the reserved CPU count.
+
+When two or more devices are specified, the net queues count is set to any net device that matches one of them.
+====
++
+An example performance profile configuration is shown below:
++
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+  kind: PerformanceProfile
+  metadata:
+    name: manual
+  spec:
+    cpu:
+    isolated: 3-51,54-103
+    reserved: 0-2,52-54
+
+    net:
+    - userLevelNetworking: true         <1>
+    #!
+
+    more examples:
+
+      net:
+        userLevelNetworking: true
+        devices:
+            -interfaceName: “eth0”
+            -interfaceName: “eth1”      <2>
+            -vendorID: “0x1af4”
+             deviceID: “0x1000”
+
+      net:
+       userLevelNetworking: true        <3>
+       devices:
+        -interfaceName: “eth*”
+
+      net:
+       userLevelNetworking: true        <4>
+       devices:
+        -interfaceName: “!eno1”
+
+     net:
+       userLevelNetworking: true        <5>
+       devices:
+        -interfaceName: “eth0”
+         vendorID: “0x1af4”
+         deviceID: “0x1000”
+
+    #!
+
+      nodeSelector:
+          node-role.kubernetes.io/worker-cnf: ""
+
+----
+Set the queue count to the reserved CPU count for:
+<1>  All devices.
+<2>  All devices that match any of the defined device identifiers.
+<3>  All devices starting with the interface name `eth`.
+<4>  All devices with an interface named anything other than `eno1`.
+<5>  All devices that have an interface name `eth0`, `vendorID` 0x1af4, and `deviceID` 0x1000.
+
+. Apply the performance profile.
++
+[source,terminal]
+----
+$  oc apply -f <your_profile_name>.yaml
+----
+
+[id="verify-queue-status_{context}"]
+== Verify the queue status
+
+In this section, a number of examples illustrate different performance profiles and how to verify the changes are applied.
+
+.Example 1
+
+In this example, the net queue count is set to the reserved CPU count (2) for _all_ supported devices.
+
+The relevant section from the performance profile is:
+
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+metadata:
+ name: performance
+spec:
+  kind: PerformanceProfile
+  spec:
+   cpu:
+    reserved: 0-1  #total = 2
+    Isolated: 2-8
+   net:
+    userLevelNetworking: true
+    [...]
+----
+
+Display the status of the queues associated with a device using the command:
```

Comment:
Should this be a step or a bullet?

Reply:
Yeah these ones are tough because they are examples, and not quite a procedure. Probably could be made into a procedure, but I think I've seen other places in our docs do examples that don't quite fit one of our module types. So I don't really know :)